enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,437 @@
+    * UNTIL the schema exactly matches the version we provide we set `Upgradable=false`
+2. Once CRDs are not present OR are an exact match we set `Upgradable=true`
+
+> **Note**: A **cluster-admin** is required for these steps.
+
+> **Note**: The logic for these lives in in the previous release (4.18), but
+> does not need to be carried forward to future releases as other logic exists
+> there to handle Gateway API CRD state (see below).
+
+### Post-upgrade
+
+> **Note**: The logic for these lives in in the new release (4.19) and onward.
+
+1. The CIO is hereafter deployed alongside its CRD protection [Validating Admission Policy (VAP)]
+2. The CIO constantly checks for the presence of the CRDs
+  * IF the CRDs are present
+    * UNTIL the CRD schema matches what is expected, the CIO upgrades them
+    * IF the upgrade fails `Degraded` status is set
+  * ELSE the CRDs are deployed by the CIO
+
+> **Note**: If we reach `Degraded` its expected that some tampering has
+> occurred (e.g. a cluster-admin has for some reason destroyed our VAP and
+> manually changed the CRDs). For the initial release we will simply require
+> manual intervention (support) to fix this as we can't guess too well at the
+> original intent behind the change. In future iterations we may consider more
+> solutions if this becomes a common problem for some reason.
+
+[Validating Admission Policy (VAP)]:https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/
+
+### API Extensions
+
+None.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Hypershift runs the Ingress Operator on the management cluster but configured
+with a kubeconfig to manage resources on the guest cluster.  This means that the
+Ingress Operator manages the Gateway API CRDs on the guest cluster, the same as
+on standalone clusters.
+
+#### Standalone Clusters
+
+For standalone clusters, the Ingress Operator manages the Gateway API CRD
+life-cycle so that Gateway API can be configured and used post-install.
+
+#### Single-node Deployments or MicroShift
+
+The Cluster Ingress Operator (CIO) will be used to deploy and manage the
+lifecycle of the Gateway API CRDs for applicable clusters. CRDs and their
+management are not particularly resource intensive.
+
+The CIO covers Single Node Deployments. MicroShift however, does not run the
+CIO has its own design for supporting Gateway API (see the [MicroShift Gateway
+API Support Enhancement]).
+
+[MicroShift Gateway API Support Enhancement]:../microshift/gateway-api-support.md
+
+### Implementation Details/Notes/Constraints
+
+> What are some important details that didn't come across above in the
+> **Proposal**? Go in to as much detail as necessary here. This might be
+> a good place to talk about core concepts and how they relate. While it is useful
+> to go into the details of the code changes required, it is not necessary to show
+> how the code will be rewritten in the enhancement.
```

Comment:
You can just add a TBD here for now.

Reply:
```suggestion
> What are some important details that didn't come across above in the
> **Proposal**? Go in to as much detail as necessary here. This might be
> a good place to talk about core concepts and how they relate. While it is useful
> to go into the details of the code changes required, it is not necessary to show
> how the code will be rewritten in the enhancement.

TBD
```